src/events.js

```diff
@@ -1377,7 +1383,7 @@ const collectionSubmission = async (participantData, biospecimenData, cntd) => {
             focus = false;
         }
 
-        if (input.required) biospecimenData[`${input.id.replace('Id', '')}`][conceptIds.collection.tube.scannedId] = `${masterID} ${tubeID}`.trim();
+        if (input.required) biospecimenData[tubeConceptId][conceptIds.collection.tube.scannedId] = `${masterID} ${tubeID}`.trim();
```

Comment:
should we check if `biospecimenData[tubeConceptId]` exists

Reply:
Thanks, I added this check on 1348 & 1386.